shade_janitor/cleanup.py

```diff
@@ -1,65 +1,136 @@
 #!/usr/bin/env python
 
+def cleanup_instances(instances):
+    """Cleanup instances."""
+    for uuid in instances:
+        cloud.delete_server(uuid, wait=True, delete_ips=True)
+
+def dry_cleanup_instances(instances):
+    """Dry cleanup of instances."""
+    for uuid in instances:
+        print('nova delete {}'.format(uuid))
+
+def dry_cleanup_router_interfaces(routers_id, subnets_id):
+    """Dry cleanup of router interfaces."""
+    for r_uuid in routers_id:
+        for uuid in subnets_id:
+            print('neutron router-interface-delete {0} {1}'.format(
+                r_uuid, uuid))
+
+def cleanup_router_interfaces(routers_id, subnets_id):
+    """Cleanup router interfaces."""
+    for r_uuid in routers_id:
+        for uuid in subnets_id:
+            for router in cloud.search_routers(r_uuid):
+                cloud.remove_router_interface(router, uuid)
+
+def setup_router_interfaces_cleanup(routers, router_interfaces):
+    """Setup router interfaces cleanup."""
```

Comment:
This is a brute force approach, to router interface cleanup. We might want to make note here of that.